backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/KonanConfig.kt

```diff
@@ -145,6 +145,24 @@ class KonanConfig(val project: Project, val configuration: CompilerConfiguration
         getIncludedLibraries(includedLibraryFiles, configuration, resolvedLibraries)
     }
 
+    val usedDynamicCaches: List<String>
+        get() = configuration.get(KonanConfigKeys.USED_DYNAMIC_CACHES)!!
+
+    val usedStaticCaches: List<String>
+        get() = configuration.get(KonanConfigKeys.USED_STATIC_CACHES)!!
+
+    // TODO: is "featured libraries" concept applicable here?
+    //   cache may contain more libraries than required,
+    //   and it may be convenient to pass all these libraries as -Xcached-library.
+    val cachedLibraries by lazy {
+        val cachedLibrariesFiles = configuration.get(KonanConfigKeys.CACHED_LIBRARIES)!!.toSet()
+        val usesCache = usedDynamicCaches.isNotEmpty() || usedStaticCaches.isNotEmpty()
+        resolvedLibraries.getFullList().filter {
+            usesCache && it.isDefault // For simplicity; default libraries are anyway included to cache by default.
```

Comment:
Not sure, if this is good idea. They could be big, and we shall only include them if we were linking them before.

Reply:
Also note, that some platform libraries may have non-trivial initializers, so this approach will bring a lot of uneeded init unconditionally.

Reply:
> Not sure, if this is good idea.

Well, there are at least two reasons behind this decision.
1. Default libraries are included implicitly. For example, Gradle is unaware of these ones. So any other cache implementation would require compiler to be more complicated in handling default libraries.
2. Consider the case of managing the cache manually. There are quite a lot of platform libraries on Apple platforms, and figuring out which are demanded (and thus should be cached too) requires applying some efforts.

> They could be big, and we shall only include them if we were linking them before.

After migrating interop libraries to metadata this shouldn't be such an issue.

> some platform libraries may have non-trivial initializers

Not sure if this is correct.
Platform libraries are interop libraries. Interop libraries don't have non-trivial Kotlin initializers (and must not have them).